internal/services/dispatcher/server.go

```diff
@@ -492,6 +492,54 @@ func (s *DispatcherImpl) ListenV2(request *contracts.WorkerListenRequest, stream
 		s.workers.DeleteForSession(request.WorkerId, sessionId)
 	}()
 
+	// HOTFIX: Start a background goroutine to poll and ensure worker stays active
+	// This addresses the race condition where multiple listeners can interfere with each other's active status
+	//
+	// Race condition scenario:
+	// 1. Worker A establishes ListenV2 connection, sets worker active=true with timestamp T1
```

Comment:
I would update this comment to indicate that it is the same worker that's establishing the connection, this is incorrect as written